docs/advanced/snapshot.mdx

```diff
@@ -1,7 +1,24 @@
 ---
-title: 'Snapshot'
+title: 'snapshot'
 section: 'Advanced'
 description: 'Create a snapshot of current state'
 ---
 
 # snapshot
+
+#### A snapshot takes a proxy and returns an immutable object, unwrapped from the proxy. In sequential snapshot calls, when the values in the proxy have not changed, a pointer to the same object is returned. This allows for shallow comparison in render functions, preventing spurious renders. Snapshots also throw promises, making them work with React Suspense.
```

Comment:
We'd like to add a note that the immutable object is actually deeply frozen. Can be an additional note below.